base/linalg/bidiag.jl

```diff
@@ -79,22 +74,24 @@ julia> A = [1 1 1 1; 2 2 2 2; 3 3 3 3; 4 4 4 4]
  3  3  3  3
  4  4  4  4
 
-julia> Bidiagonal(A, :U) #contains the main diagonal and first superdiagonal of A
-4×4 Bidiagonal{Int64}:
+julia> Bidiagonal(A, :U) # contains the main diagonal and first superdiagonal of A
+4×4 Bidiagonal{Int64,Array{Int64,1}}:
  1  1  ⋅  ⋅
  ⋅  2  2  ⋅
  ⋅  ⋅  3  3
  ⋅  ⋅  ⋅  4
 
-julia> Bidiagonal(A, :L) #contains the main diagonal and first subdiagonal of A
-4×4 Bidiagonal{Int64}:
+julia> Bidiagonal(A, :L) # contains the main diagonal and first subdiagonal of A
+4×4 Bidiagonal{Int64,Array{Int64,1}}:
  1  ⋅  ⋅  ⋅
  2  2  ⋅  ⋅
  ⋅  3  3  ⋅
  ⋅  ⋅  4  4
 ```
 """
-Bidiagonal(A::AbstractMatrix, uplo::Symbol) = Bidiagonal(diag(A), diag(A, uplo == :U ? 1 : -1), uplo)
+function Bidiagonal(A::AbstractMatrix, uplo::Symbol)
+    Bidiagonal(diag(A, 0), diag(A, uplo == :U ? 1 : -1), uplo)
```

Comment:
Yeah, as long as the two arg form is inferable this should be "more correct" since `diag(A)` in some cases give something else.